### PR TITLE
feat(server): send email

### PR DESCRIPTION
**Pull request by DavidLiang01**
To test the send email feature, add .env file in the server folder with EMAIL_PASS as key. For the password, please message me.

You can use the following as a body in the request when calling the endpoint:
```
{
    "to": ["soen390.team07@gmail.com"],
    "subject": "test",
    "emailBody": "heyee"
}
```

### CODE REVIEW

File: server/app/services/UserService.ts [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
 @injectable()
 export class UserService {
+  private transporter: Mail;
   @inject(TYPES.config) private config: IConfig;
-  constructor(@inject(TYPES.UserRepository) private userRepo: UserRepository) {}
+  constructor(@inject(TYPES.UserRepository) private userRepo: UserRepository) {
+    this.transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: 'soen390.team07@gmail.com',
```
Comment (Mackbellemore) [inline server/app/services/UserService.ts:19]: store this in the default.js file

Comment (DavidLiang01) [reply server/app/services/UserService.ts:19]: ok

File: server/app/controllers/UserController.ts [hunk 2]
```diff
@@ -40,4 +41,14 @@ export class UserController extends BaseHttpController {
       return this.json(err.message, 400);
     }
   }
+
+  @httpPost('/email')
```
Comment (Mackbellemore) [inline server/app/controllers/UserController.ts:45]: im wondering if we should maybe just create a new controller, so the path can just be /email

File: server/app/services/UserService.ts [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
+import nodemailer, { SentMessageInfo } from 'nodemailer';
+import Mail from 'nodemailer/lib/mailer';
 
 @injectable()
 export class UserService {
+  private transporter: Mail;
   @inject(TYPES.config) private config: IConfig;
-  constructor(@inject(TYPES.UserRepository) private userRepo: UserRepository) {}
+  constructor(@inject(TYPES.UserRepository) private userRepo: UserRepository) {
+    this.transporter = nodemailer.createTransport({
```
Comment (Mackbellemore) [inline server/app/services/UserService.ts:16]: I would add this nodemailer.createTransport call to the registery.ts file so you can just inject constant value like we do for a logger

Comment (Mackbellemore) [reply server/app/services/UserService.ts:16]: example: https://github.com/Mackbellemore/soen-390-team07/blob/develop/server/app/registry.ts#L27

File: server/app/services/UserService.ts [hunk 4]
```diff
@@ -35,4 +46,16 @@ export class UserService {
 
     return UserEntity.buildUser(user);
   }
+
+  public async sendEmail(body: IEmail): Promise<SentMessageInfo> {
+    const emailList = body.to.join(', ');
+
+    const emailDetails = {
+      from: 'no-reply@ERPSystem TEAM07 <soen390.team07@gmail.com>',
```
Comment (Mackbellemore) [inline server/app/services/UserService.ts:54]:
```suggestion
      from: `no-reply@ERPSystem TEAM07 <${config.get<string>(email)}>`,
```

File: server/app/controllers/UserController.ts (hunk 2 above)
Comment (DavidLiang01) [reply server/app/controllers/UserController.ts:45]: I was thinking the same, I can create a controller for the system such as /system/email. What do you think?

File: server/app/services/UserService.ts (hunk 3 above)
Comment (DavidLiang01) [reply server/app/services/UserService.ts:16]: great! I will make the changes

Comment (Mackbellemore) [reply server/app/services/UserService.ts:16]: Also I would create a EmailService.ts to have this seperately along with a EmailController.ts file

File: server/app/controllers/UserController.ts (hunk 2 above)
Comment (Mackbellemore) [reply server/app/controllers/UserController.ts:45]: ya that sounds good, i dont think there should be anything email related in the user controller

File: server/config/default.js [hunk 5]
```diff
@@ -23,5 +23,6 @@ module.exports = {
     secret: process.env.JWT_SECRET || 'addadasjfghKEU10dawadrfgh!e29sosafelol',
     expiry: process.env.JWT_EXPIRY || '30m',
   },
+  mail: 'ERP_Soen390-team07',
```
Comment (Mackbellemore) [inline server/config/default.js:26]: if this is the password, I would'nt expose this to github since this is a legit email account

Comment (DavidLiang01) [reply server/config/default.js:26]: I was thinking the same, I am looking for a solution for that. Any suggestions?

**Comment by gordonpn** [comment]
Ima review tomorrow

File: server/config/default.js (hunk 5 above)
Comment (Mackbellemore) [reply server/config/default.js:26]:
```suggestion
  mail: 'process.env.EMAIL_PASSWORD',
```

Then create a .env file and store the password there.
EMAIL_PASSWORD='ERP_Soen390-team07'

File: server/app/services/SystemService.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { IEmail } from './../models/UserModel';
+import { inject, injectable } from 'inversify';
+import TYPES from '../constants/types';
+import config from 'config';
+import { SentMessageInfo } from 'nodemailer';
+import Mail from 'nodemailer/lib/mailer';
+import dotenv from 'dotenv';
```
Comment (Mackbellemore) [inline server/app/services/SystemService.ts:7]: this isnt needed if youre using the config

File: server/app/registry.ts [hunk 7]
```diff
@@ -32,11 +36,20 @@ logger.add(
   })
 );
 
+const transporter = nodemailer.createTransport({
+  service: 'gmail',
+  auth: {
+    user: config.get<string>('email'),
+    pass: process.env.EMAIL_PASS,
```
Comment (Mackbellemore) [inline server/app/registry.ts:43]: add this in the default.js file and use config.get<string>('pass')

Comment (DavidLiang01) [reply server/app/registry.ts:43]: do you mean for line 43?

Comment (Mackbellemore) [reply server/app/registry.ts:43]: ya dont call process.env directly, use config like we do everywhere else

**Review by Mackbellemore — APPROVED** [review]

**Review by gordonpn — APPROVED** [review]
Good job David

**Pull request merged**